criu/mount.c

```diff
@@ -1723,38 +1723,49 @@ struct mount_info __maybe_unused *add_cr_time_mount(struct mount_info *root, cha
 	return NULL;
 }
 
-/* Returns 1 in case of success, -errno in case of mount fail, and 0 on other errors */
+/*
+ * Returns:
+ *  0 - success
+ * -1 - error
+ *  1 - skip
+ */
 static __maybe_unused int mount_cr_time_mount(struct ns_id *ns, unsigned int *s_dev, const char *source,
 					      const char *target, const char *type)
 {
-	int mnt_fd, cwd_fd, ret, exit_code = 0;
+	int mnt_fd, cwd_fd, exit_code = -1;
 	struct stat st;
 
-	ret = switch_mnt_ns(ns->ns_pid, &mnt_fd, &cwd_fd);
-	if (ret < 0) {
+	if (switch_mnt_ns(ns->ns_pid, &mnt_fd, &cwd_fd)) {
 		pr_err("Can't switch mnt_ns\n");
-		goto out;
+		return -1;
 	}
 
-	ret = mount(source, target, type, 0, NULL);
-	if (ret < 0) {
-		pr_perror("Unable to mount %s %s", source, target);
-		exit_code = -errno;
-		goto restore_ns;
-	} else {
-		if (stat(target, &st) < 0) {
-			pr_perror("Can't stat %s", target);
-			exit_code = 0;
-		} else {
-			*s_dev = MKKDEV(major(st.st_dev), minor(st.st_dev));
+	if (mount(source, target, type, 0, NULL)) {
+		switch (errno) {
+		case EPERM:
+		case EBUSY:
+		case ENODEV:
+		case ENOENT:
+			pr_debug("Skipping binfmt_misc as was unable to mount it: %m\n");
```

Comment:
* s/binfmt_misc/%s...", type
* %m can't be used here, it has to be replaced with %s ...", strerror(errno)

Reply:
done